src/protosym/core/evaluate.py

```diff
@@ -26,30 +26,91 @@
 
 
 if _TYPE_CHECKING:
-    from typing import Optional, Iterable
+    from typing import Optional, Iterable, Any
 
     Op1 = Callable[[_T], _T]
     Op2 = Callable[[_T, _T], _T]
     OpN = Callable[[Iterable[_T]], _T]
 
 
+def _generic_operation_error(head: TreeExpr, argvals: Iterable[_T]) -> _T:
```

Comment:
Is this `_T` type hint for the return correct if an error is being raised before implicit `None` would be returned?

Reply:
Raising an exception is considered separate to the return type of a function. The contract is that the function must either return the specified type or raise an exception. In this case we always raise an exception so the type could be documented as anything but I made it consistent with what is expected in the type hint for `add_op_generic`.

I have seen several discussions about making exceptions part of typing but they all stumble on the fact that basically anything in Python can raise an exception.

Reply:
That makes it clear what it's the way that it is. Thank you for explaining.